R/conditional-transform.R

```diff
@@ -0,0 +1,215 @@
+#' # of the categories that are in variables Opinion1, Opinion2, or Opinion3, 
+#' # just those levels that are selected by the conditions specified).
+#' 
+#' ds$days_with_dog <- conditionalTransform(pet1 == 'Cat' ~ Opinion1,
+#'                                          pet2 == 'Cat' ~ Opinion4,
+#'                                          pet3 == 'Cat' ~ Opinion3,
+#'                                          data = ds, type = "categorical",
+#'          categories = c("Strongly Agree", "Agree", "Neither Agree Nor Disagree",
+#'          "Disagree", "Strongly Disagree"))
+#' 
+#' as.vector(ds$cat_opinion)
+#' # [1] "Strongly Agree" "Disagree" "Agree" "Strongly Disagree" ...
+#' # Levels: Strongly Agree  Agree  Neither Agree Nor Disagree  Disagree  Strongly Disagree
+#'                                        
+#' # We can also use `conditionalTransform` to return a string as well as the
+#' # contents of other variables, if for example we want to separate out people
+#' # who have only had pets for less than a year, we can specify that as the
+#' # first condition (which will be used even if the subsequent conditions are
+#' # also true).
+#' 
+#' ds$dog_opinion <- conditionalTransform(days_having_pet < 365 ~ "too early to tell",
+#'                                        pet1 == 'Dog' ~ Opinion1,
+#'                                        pet2 == 'Dog' ~ Opinion2,
+#'                                        pet3 == 'Dog' ~ Opinion3,
+#'                                        data = ds)
+#' as.vector(ds$dog_opinion)
+#' # [1] "Strongly Agree" "Disagree" "too early to tell" "Strongly Disagree"
+#' # Levels: Agree  Disagree  Strongly Agree  Strongly Disagree  too early to tell
+#'                                     
+#' # Further, we can also use conditional transform with numeric source variables.
+#' 
+#' ds$days_with_dog <- conditionalTransform(pet1 == 'Dog' ~ days_having_pet,
+#'                                          pet2 == 'Dog' ~ days_having_pet,
+#'                                          pet3 == 'Dog' ~ days_having_pet,
+#'                                          data = ds, type = "numeric")
+#' as.vector(ds$days_with_dog)
+#' # [1] 300 57 70 5 ...
+#' 
+#' # conditionalTransform will evaluate is.na() or NA as conditoins or sources 
+#' # respectively
+#' 
+#' ds$cat_opinion <- conditionalTransform(pet1 == 'Cat' ~ Opinion1,
+#'                                        pet2 == 'Cat' ~ Opinion2,
+#'                                        pet3 == 'Cat' ~ Opinion3, 
+#'                                        is.na(pet1) & is.na(pet2) & is.na(pet3) ~ "None",
+#'                                        data = ds)
+#' as.vector(ds$cat_opinion)
+#' # [1] "Strongly Agree" "Disagree" "Agree" "Strongly Disagree" "None" ...
+#' # Levels: Agree  Disagree None Strongly Agree  Strongly Disagree
+#' 
+#' 
+#' # Finally, when constructing conditionalTransforms, it is frequently helpful 
```

Comment:
This is reading to me more like a vignette than a man page. That's fine, it's easy to link a man page to a vignette.